src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java

```diff
@@ -322,6 +330,16 @@ private Mono<Optional<List<RequestedEquipmentTO>>> createRequestedEquipmentsByBo
       return Mono.just(Optional.of(Collections.emptyList()));
     }
 
+    if (requestedEquipments.stream()
```

Comment:
mixes streams and Reactor. Suggest rewriting this with Reactor only, that way in Reactive Stream can be used consolidating this validation, the mapping and the saving.